app/src/main/java/org/gdg/frisbee/android/api/OkClientProvider.java

```diff
@@ -0,0 +1,45 @@
+package org.gdg.frisbee.android.api;
+
+import android.content.Context;
+
+import com.squareup.okhttp.Cache;
+import com.squareup.okhttp.OkHttpClient;
+
+import java.io.File;
+
+import retrofit.client.Client;
+import retrofit.client.OkClient;
+
+/**
+ * Created by <a href="mailto:marcusandreog@gmail.com">Marcus Gabilheri</a>
+ *
+ * @author Marcus Gabilheri
+ * @version 1.0
+ * @since 7/26/15.
+ */
+public final class OkClientProvider {
+
+    // Cache size for the OkHttpClient
+    private static final int DISK_CACHE_SIZE = 50 * 1024 * 1024; // 50MB
+
+    private static OkClient okClient;
+
+    private OkClientProvider() {
+    }
+
+    public static OkHttpClient getOkHttpClient(Context context) {
+        OkHttpClient okHttpClient = new OkHttpClient();
+        // Install an HTTP cache in the application cache directory.
+        File cacheDir = new File(context.getApplicationContext().getCacheDir(), "http");
+        Cache cache = new Cache(cacheDir, DISK_CACHE_SIZE);
```

Comment:
I think it should be a singleton. But that doesn't mean that it should be
static. There should one be one instance. And that would be an instance in
App class. Or we can remove it from the app class and have a static
singleton instance. But accessing it through App instance and making it a
static seems too much to me.

On Tue, Jul 28, 2015, 9:12 PM Marcus Gabilheri <notifications@github.com>
wrote:

> In app/src/main/java/org/gdg/frisbee/android/api/OkClientProvider.java
> <https://github.com/gdg-x/frisbee/pull/432#discussion_r35680316>:
>
> > + */
> > +public final class OkClientProvider {
> > +
> > +    // Cache size for the OkHttpClient
> > +    private static final int DISK_CACHE_SIZE = 50 * 1024 * 1024; // 50MB
> > +
> > +    private static OkClient okClient;
> > +
> > +    private OkClientProvider() {
> > +    }
> > +
> > +    public static OkHttpClient getOkHttpClient(Context context) {
> > +        OkHttpClient okHttpClient = new OkHttpClient();
> > +        // Install an HTTP cache in the application cache directory.
> > +        File cacheDir = new File(context.getApplicationContext().getCacheDir(), "http");
> > +        Cache cache = new Cache(cacheDir, DISK_CACHE_SIZE);
>
> Why not just have one instance as a singleton? Jake Wharton always says on
> his talks that OkHttpClient is design to be used as a singleton.
> Here's a reference where he mentions it (
> https://plus.google.com/118239425803358296962/posts/5nzAvPaitHu).
>
> He also always uses OkHttpClient as singleton on the apps he open sources
> such as u2020.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/gdg-x/frisbee/pull/432/files#r35680316>.
>